website/pages/docs/secrets/databases/custom.mdx

```diff
@@ -101,17 +186,96 @@ You should now be able to register your plugin into the vault catalog. To do
     sha256="..." \
-    command="myplugin"
-Success! Data written to: sys/plugins/catalog/database/myplugin-database-plugin
+    command="mydatabase"
+Success! Data written to: sys/plugins/catalog/database/mydatabase-database-plugin
 ```
 
 Now you should be able to configure your plugin like any other:
 
 ```shell-session
-$ vault write database/config/myplugin \
-    plugin_name=myplugin-database-plugin \
+$ vault write database/config/mydatabase \
+    plugin_name=mydatabase-database-plugin \
     allowed_roles="readonly" \
     myplugins_connection_details="..."
 ```
+
+## Upgrading database plugins
+
+### Background
+
+In Vault 1.6, the database interface changed. The new version is referred to as version 5
+and the previous version as version 4. This is due to prior versioning of the interface
+that was not explicitly exposed.
+
+The new interface was introduced for several reasons:
+
+1. [Password policies](/docs/concepts/password-policies) introduced in Vault 1.5 required
+   that Vault be responsible for generating passwords. In the prior version, the database
+   plugin was responsible for generating passwords. This prevented integration with
+   password policies.
+2. Passwords needed to be generated by database plugins. This meant that plugin authors
+   were responsible for generating secure passwords. This should be done with a helper
+   function available within the Vault SDK, however there was nothing preventing an
+   author from generating insecure passwords.
+3. There were a number of inconsistencies within the version 4 interface that made it
+   confusing for authors. For instance: passwords were handled in 3 different ways.
+   `CreateUser` generated a password and returned it, `SetCredentials` receives a password
+   via a configuration struct and returns it, and `RotateRootCredentials` generated a
+   password and was expected to return an updated copy of its entire configuration
+   with the new password.
+4. The `SetCredentials` and `RotateRootCredentials` used for static credential rotation,
+   and rotating the root user's credentials respectively were essentially the same operation:
+   change a user's password. The only practical difference was which user it was referring
+   to. This was especially evident when `SetCredentials` was used when rotating root
+   credentials (unless static credential rotation wasn't supported by the plugin in question).
+
+The new interface is roughly modeled after a [gRPC](https://grpc.io/) interface. It has improved
+future compatibility by not requiring changes to the interface definition to add additional data
+in the requests or responses. It also simplifies the interface by merging several into a single
+function call.
+
+### Upgrading your custom database
+
+Vault 1.6 supports both version 4 and version 5 database plugins. The support for version 4
+plugins will be removed in a future release. If you are using a version 4 custom database
```

Comment:
Although it's implied in the instructions below, I think it might be helpful for users if we explicitly tell them how to determine whether their plugin is version 4 or 5 (i.e. check the `sdk` import paths used in its source, A or B)

Reply:
Or alternatively, augment some of the references to the fairly abstract "version 4" and "version 5" with concrete references to the interfaces implemented.

Reply:
Good call. I've added a small section describing some ways that a user can tell if they are on v4 or v5 as well as references to the files where the interface definitions are.